instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java

```diff
@@ -28,7 +29,12 @@ public void consumeMessageBefore(ConsumeMessageContext context) {
       return;
     }
     Context parentContext = Context.current();
-    Context newContext = instrumenter.start(parentContext, context.getMsgList());
+
+    String consumerGroup = context.getConsumerGroup();
+    Baggage baggage = Baggage.fromContext(parentContext);
+    Baggage newBaggage = baggage.toBuilder().put("consumerGroup", consumerGroup).build();
```

Comment:
Using baggage to pass info is really roundabout, you could change the type of the request that is passed to `instrumenter.start` to an object that contains both message list and consumer group. For example see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c166c937b3417dbc6bfcaafd7321483fcf0c0c37/instrumentation/spring/spring-integration-4.1/library/src/main/java/io/opentelemetry/instrumentation/spring/integration/TracingChannelInterceptor.java#L73 message and message channel are packed into `MessageWithChannel` that is used as request.

Reply:
Are you talking about adding a custom RocketMQ interceptor to implement the functionality in the same way that Spring implements interceptors?

Reply:
No, I meant that you could create a class `RocketMqRequest` that contains a `MessageExt` and consumer group. In `RocketMqConsumerInstrumenter` change the instrumenters that are parametrized over `MessageExt` to be parametrized over `RocketMqRequest`. This way you should be able to get consumer group in `RocketMqConsumerAttributeGetter` or `RocketMqConsumerExperimentalAttributeExtractor`.

Reply:
OK，Let me try，Thank you。